Doc/constraints.txt

```diff
@@ -0,0 +1,24 @@
+# We have upper bounds on our transitive dependencies here
+# To avoid new releases unexpectedly breaking our build.
+# This file can be updated on an ad-hoc basis,
+# though it will probably have to be updated
+# whenever Doc/requirements.txt is updated.
+
+# Direct dependencies of Sphinx
+babel<3
+colorama<0.5
+imagesize<1.5
+Jinja2<3.2
```

Comment:
```suggestion
```
this constraint is superfluous to `jinja2==3.0.3` from `requirements.txt`

Reply:
For the sake of minimalism, we could drop or modify the constraint to be in line with the pin, otherwise it will be confusing IMHO.

Reply:
3.12 branch is using Sphinx 4.5.0 whereas 3.10 uses Sphinx 2.4.4 and Jinja constraints are different for those versions on logical/technical side of things.